llvm/lib/Target/NVPTX/NVPTXAssignValidGlobalNames.cpp

```diff
@@ -73,10 +74,10 @@ std::string NVPTXAssignValidGlobalNames::cleanUpName(StringRef Name) {
   std::string ValidName;
   raw_string_ostream ValidNameStream(ValidName);
   for (char C : Name) {
-    if (C == '.' || C == '@' || C == '<' || C == '>') {
-      ValidNameStream << "_$_";
-    } else {
+    if (isAlnum(C) || C == '_' || C == '$') {
```

Comment:
PTX also allows '%' as the first character:
https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#identifiers
> followsym:   [a-zA-Z0-9_$]
> identifier:  [a-zA-Z]{followsym}* | {[_$%]{followsym}+

Reply:
LLVM will throw a fatal error for '%' in symbol names in `MCSymbol::print`

Reply:
That's worth a comment explaining the discrepancy between what PTX allows and what we're checking for here.

Reply:
Sounds good, added